lib/screens/home_screen.dart

```diff
@@ -1,39 +1,195 @@
+// Copyright 2021 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
 import 'package:flutter/material.dart';
+import 'package:galacticcoders_website/screens/about_us_screen.dart';
+import 'package:galacticcoders_website/screens/blog_screen.dart';
+import 'package:galacticcoders_website/screens/component_screen.dart';
+import 'package:galacticcoders_website/screens/contact_us_screen.dart';
+import 'package:galacticcoders_website/screens/projects_screen.dart';
+import 'package:galacticcoders_website/screens/events_screen.dart';
+
+void main() {
+  runApp(const HomeScreen());
+}
 
-class HomeScreen extends StatelessWidget {
+class HomeScreen extends StatefulWidget {
   const HomeScreen({super.key});
 
+  @override
+  State<HomeScreen> createState() => _HomeScreenState();
+}
+
+// NavigationRail shows if the screen width is greater or equal to
+// screenWidthThreshold; otherwise, NavigationBar is used for navigation.
+const double narrowScreenWidthThreshold = 450;
+
+const Color m3BaseColor = Color(0xff6750a4);
+const List<Color> colorOptions = [
+  m3BaseColor,
+  Colors.blue,
+  Colors.teal,
+  Colors.green,
+  Colors.yellow,
+  Colors.orange,
+  Colors.pink
+];
+const List<String> colorText = <String>[
+  "M3 Baseline",
+  "Blue",
+  "Teal",
+  "Green",
+  "Yellow",
+  "Orange",
+  "Pink",
+];
+
+class _HomeScreenState extends State<HomeScreen> {
+  bool useLightMode = true;
+  int screenIndex = 0;
+
+  late ThemeData themeData;
+
+  @override
+  initState() {
+    super.initState();
+    themeData = updateThemes(useLightMode);
+  }
+
+  ThemeData updateThemes(bool useLightMode) {
+    return ThemeData(
+        colorSchemeSeed: Colors.blue,
+        useMaterial3: true,
+        brightness: useLightMode ? Brightness.light : Brightness.dark);
+  }
+
+  void handleScreenChanged(int selectedScreen) {
+    setState(() {
+      screenIndex = selectedScreen;
+    });
+  }
+
+  void handleBrightnessChange() {
+    setState(() {
+      useLightMode = !useLightMode;
+      themeData = updateThemes(useLightMode);
+    });
+  }
+
+  void handleMaterialVersionChange() {
+    setState(() {
+      themeData = updateThemes(useLightMode);
+    });
+  }
+
+  void handleColorSelect(int value) {
+    setState(() {
+      themeData = updateThemes(useLightMode);
+    });
+  }
+
+  Widget createScreenFor(int screenIndex, bool showNavBarExample) {
+    switch (screenIndex) {
+      case 0:
+        return const AboutUsScreen();
+      case 1:
+        return const ProjectsScreen();
+      case 2:
+        return const EventsScreen();
+      case 3:
+        return const BlogScreen();
+      case 4:
+        return const ContactUsScreen();
+      default:
+        return const AboutUsScreen();
+    }
+  }
+
+  PreferredSizeWidget createAppBar() {
```

Comment:
Excellent! Let me send you this discussion regarding "Should I use methods to create widgets?" The short answer is "no", preferably not.

https://stackoverflow.com/questions/69350472/methods-vs-separate-classes-for-widgets-in-flutter

In Flutter it is considered an anti-pattern. Unnecessary rebuilds will be triggered AND the widgets might not correctly retrieve their context - crazy huh?

I would recommend turning this method [createAppBar] into a stateless widget of it's own.

Reply:
Oh, here's a small blog post about it as well: https://iiro.dev/splitting-widgets-to-methods-performance-antipattern/